src/test/scala/org/constellation/consensus/RoundManagerTest.scala

```diff
@@ -0,0 +1,286 @@
+  BroadcastUnionBlockProposal,
+  RoundInfo
+}
+import org.constellation.primitives.Schema.{NodeType, SignedObservationEdge}
+import org.constellation.primitives._
+import org.constellation.primitives.storage.CheckpointService
+import org.mockito.integrations.scalatest.IdiomaticMockitoFixture
+import org.scalatest.{BeforeAndAfter, FunSuiteLike, Matchers, OneInstancePerTest}
+
+import scala.collection.concurrent.TrieMap
+import scala.concurrent.duration._
+
+class RoundManagerTest
+    extends TestKit(ActorSystem("RoundManagerTest"))
+    with FunSuiteLike
+    with Matchers
+    with IdiomaticMockitoFixture
+    with BeforeAndAfter
+    with OneInstancePerTest {
+
+  implicit val dao: DAO = mock[DAO]
+  implicit val materialize: ActorMaterializer = ActorMaterializer()
+
+  val roundManagerProbe = TestProbe()
+  val roundManager: TestActorRef[RoundManager] =
+    TestActorRef(Props(spy(new RoundManager(60 seconds))), roundManagerProbe.ref)
+
+  val checkpointFormationThreshold = 1
+  val daoId = Schema.Id("a")
+
+  val facilitatorId1 = Schema.Id("b")
+  val peerData1 = mock[PeerData]
+  peerData1.peerMetadata shouldReturn mock[PeerMetadata]
+  peerData1.peerMetadata.id shouldReturn facilitatorId1
+  peerData1.notification shouldReturn Seq()
+
+  val facilitatorId2 = Schema.Id("c")
+  val peerData2 = mock[PeerData]
+  peerData2.peerMetadata shouldReturn mock[PeerMetadata]
+  peerData2.peerMetadata.id shouldReturn facilitatorId2
+  peerData2.notification shouldReturn Seq()
+
+  val readyFacilitators = Map(facilitatorId1 -> peerData1, facilitatorId2 -> peerData2)
+
+  dao.keyPair shouldReturn Fixtures.tempKey
+
+  val tx1 = Fixtures.dummyTx(dao)
+  val tx2 = Fixtures.dummyTx(dao)
+
+  val soe = mock[SignedObservationEdge]
+  soe.baseHash shouldReturn "abc"
+  val tips = (Seq(soe), readyFacilitators)
+
+  dao.id shouldReturn daoId
+  dao.minCheckpointFormationThreshold shouldReturn checkpointFormationThreshold
+  dao.pullTransactions(checkpointFormationThreshold) shouldReturn Some(Seq(tx1, tx2))
+  dao.readyFacilitators() shouldReturn readyFacilitators
+  dao.peerInfo shouldReturn readyFacilitators
+  dao.pullTips(readyFacilitators) shouldReturn Some(tips)
+  dao.threadSafeMessageMemPool shouldReturn mock[ThreadSafeMessageMemPool]
+  dao.threadSafeMessageMemPool.pull(1) shouldReturn None
+  dao.checkpointService shouldReturn mock[CheckpointService]
+  dao.checkpointService.contains(*) shouldReturn true
+  dao.readyPeers(NodeType.Light) shouldReturn Map()
+
+  val peerManagerProbe = TestProbe()
+  val ipManager = mock[IPManager]
+  val peerManager = TestActorRef(Props(new PeerManager(ipManager)))
+  dao.peerManager shouldReturn peerManager
+
+  after {
+    TestKit.shutdownActorSystem(system)
+  }
+
+  test("it should not start a new block-creation round if another round is in progress") {
+    val roundInfo = mock[RoundInfo]
+    roundInfo.startedByThisNode shouldReturn true
+
+    roundManager.underlyingActor.rounds += RoundId("a") -> roundInfo
```

Comment:
You did that because you 'know' the implementation, and you shouldn't relay on that , so instead sending same message twice would be sufficient.

Reply:
don't think so. in unit tests I can know the underlaying implementation and prepare everything to behave as "one action is in progress" in that case. in e2e tests I cannot relay on the implementation and I should use it in a way you proposed

Reply:
Yes but let say for some reasons  you have multiple collection for keeping round info then you will need ad extra code to handle that in this particular example i you would send twice StartRound.. then you wouldn't have to add anything correct ?

Reply:
yes, you're right. I've changed that and figured out another issue which I've fixed too